doc/CommitStyle.md

```diff
@@ -0,0 +1,19 @@
+# Commit message style #
+
+To make the commit messages consistent, the following style of messages is recommended:
+```
+  <type>: [<section>] Your message here [Fix #<num>] (credits) ```
+with
+  * **`<type>`** Type of change, see the list below. **Always include this.**
+  * **optional `[<section>]`** if the commit is in a special section, for example graphics or build system.
```

Comment:
I wonder whether we should keep this, it's not used in the project afaik.

OpenTTD used it in branches mostly (due to SVN), although there are also [NewGRF] and [AI] things I think.